### DISCUSSION TITLE
SEO optimization, JSON schema, meta tags, sitemap_index.xml, yoast

### DISCUSSION THREAD

**Discussion by rizwan92** [wpengine/faustjs#898] [Ideas]
I am managing big news websites, i want my cients to move from wordpress hosting to healdess cms, so we could have more control over front end, having said that i need few features because it is very important for us with valid reason

JSON schema (thats really important in reducing crawling budget of google to get good rank)
meta tags (usually seo editor, and authors updates tags and descriptions)
sitemap_index.xml (my clients continuously cheack sitemap_index.xml of yoast and bought subscription of it )

i have failed ones for my clients to getting them into headldess world and use power of edge computing and global cdn, by using frontity framework they have not mentioned the best practices for seo optimization 

their decopled mode index subdomain's URL  and embeded mode increase the crawl time response
thats leads to low ranking

and most importnat thing is TTFB , TTFB decides traffic from discover

**Comment by blakewilson** [comment]
Hey @rizwan92 thanks for your question.

Have you had a chance to check out our [Sitemaps](https://faustjs.org/docs/next/guides/sitemaps) guide? This may get you started in the right direction.

In terms of meta tags, this would be done through your Next.js/React components once you've requested the data in Faust. You can set meta tags using the Next.js [`<Head>`](https://nextjs.org/docs/api-reference/next/head) component.

**Comment by rizwan92** [comment]
i am facing this issue after implementing yoast codes


```
import { handleSitemapRequests } from '@faustjs/next';
import { NextRequest, NextResponse } from 'next/server';

export default async function _middleware(req: NextRequest) {
  const sitemapRequest = await handleSitemapRequests(req, {
    wpUrl: process.env.NEXT_PUBLIC_WORDPRESS_URL,
    sitemapIndexPath: `/sitemap_index.xml`,
    sitemapPathsToIgnore: ['/author-sitemap.xml'],
    async robotsTxt(sitemapUrl) {
      return `
        User-agent: *
        Allow: /

        Sitemap: ${sitemapUrl}
      `;
    },
  });

  if (sitemapRequest) {
    return sitemapRequest;
  }

  return NextResponse.next();
}

```

<img width="319" alt="image" src="https://user-images.githubusercontent.com/19162978/173184602-52b14c47-217f-4b9f-95c1-702bb91cd054.png">

**Reply by blakewilson** [reply]
@rizwan92 Try importing `handleSitemapRequests` from `@faustjs/next/middleware`, that should solve the issue!

So:

```js
import { handleSitemapRequests } from '@faustjs/next/middleware';
```

I've added a PR to fix this in the docs: https://github.com/wpengine/faustjs/pull/901

**Comment by rizwan92** [comment]
https://topchand.com/robots.txt

by clicking link below you will see there are multiples sitemap we generate for to connect contents with googles multiple platform

User-agent: *
Allow: /
Disallow: /wp-admin/admin-ajax.php

Sitemap: https://topchand.com/sitemap_index.xml
Sitemap: https://topchand.com/sitemap-news.xml


here sitemap_index.xml used for the website contents and sitemap-news.xml used for feed to google news publishers account 

rs feed is very important and / feed urls are also important

**Reply by josephfusco** [reply]
@rizwan92, in regards to your comment about RSS feeds, Version 0.7.11 of [FaustWP](https://wordpress.org/plugins/faustwp/) is now available which prevents WordPress feed URLs from being redirected to the front-end application.

**Comment by rizwan92** [comment]
I am facing 400 responses in this below URL URL in network tab did i missed something from docs

https://publicdrop.vercel.app/posts/young-poet-ashutosh-will-recite-poetry-in-akashvani-has-already-recited-his-creations

https://publicdrop.vercel.app/_next/data/A1Gqa-KyQbo4lBhgqMOFP/posts/category/health.json

**Comment by rizwan92** [comment]
i have coded most of thing i am ready to go to productions, just stuck on few parts mentioned above

**Comment by rizwan92** [comment]
This middleware is not working in the latest next js version which is 12.2.5

**Comment by nazmateo** [comment]
Hi guys, we are also using yoast, i am facing the same issue. Is there any update on this? I am only getting { notFound: true } as response from getSitemapProps function


Thank you guys
